pms/models/pms_automated_mails.py

```diff
@@ -0,0 +1,172 @@
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class PmsAutomatedMails(models.Model):
+    _name = "pms.automated.mails"
+    _description = "Automatic Mails"
+
+    name = fields.Char(string="Name")
+
+    pms_property_id = fields.Many2one(string="Property", comodel_name="pms.property")
+
+    automated_actions_id = fields.Many2one(
+        string="Automated Actions", comodel_name="base.automation", ondelete="cascade"
+    )
+
+    time = fields.Integer(string="Time")
+
+    time_type = fields.Selection(
+        string="Time Range",
+        selection=[
+            ("minutes", "Minutes"),
+            ("hour", "Hour"),
+            ("day", "Days"),
+            ("month", "Months"),
+        ],
+        default="day",
+    )
+    template_id = fields.Many2one(
+        string="Template", comodel_name="mail.template", required=True
+    )
+
+    model_id = fields.Many2one(
+        string="Model", comodel_name="ir.model", compute="_compute_model_id", store=True
+    )
+
+    action = fields.Selection(
+        string="Action",
+        selection=[
+            ("creation", "Reservation creation"),
+            ("write", "Reservation modification"),
```

Comment:
caution with generic "write" action. Reservation is a complex model, a write action is launch in multi "insignificant" fields-